src/common/attachments/uploadAttachmentsDialog.component.tsx

```diff
@@ -0,0 +1,181 @@
+        // This should fix the multiple deletion error
+        if (id) {
+          // TODO delete if the file has been removed mid upload
+          // console.log('Delete id ', id);
+          const newMap = { ...fileMetadataMap };
+          delete newMap[file?.id ?? ''];
+        }
+
+        setIsUploading(false);
+      },
+    []
+  );
+
+  const handleClose = React.useCallback(() => {
+    onClose();
+    setFileMetadataMap({});
+    setIsUploading(false);
+  }, [onClose]);
+
+  React.useEffect(() => {
+    if (open) {
+      const uppyInstance = new Uppy<Meta, AwsBody>({
+        autoProceed: false,
+        restrictions: {
+          maxFileSize: MAX_FILE_SIZE_B,
+          requiredMetaFields: ['name'],
+        },
+      });
+
+      // Set up the S3 upload with a pre-signed URL
+      uppyInstance.use(AwsS3, {
+        shouldUseMultipart: false,
+        getUploadParameters: async (file) => {
+          // Collect metadata directly from the file object
+          const response = await postAttachmentMetadata({
+            entity_id: entityId,
+            file_name: (file.meta.name as string) || '',
+            title: file.meta.title ? (file.meta.title as string) : undefined,
+            description: file.meta.description
+              ? (file.meta.description as string)
+              : undefined,
+          });
+
+          setFileMetadataMap((prev) => ({
+            ...prev,
+            [file.id]: response.id,
+          }));
+
+          return {
+            method: 'POST',
+            url: response.upload_info.url,
+            fields: response.upload_info.fields,
+          };
+        },
+      });
+
+      // Use the FileInput and ProgressBar plugins
+
+      uppyInstance.use(ProgressBar);
+
+      setUppy(uppyInstance);
+
+      return () => uppyInstance.cancelAll();
+    }
+  }, [open, entityId, postAttachmentMetadata]);
+
+  // Track the start and completion of uploads
+  uppy?.on('upload', () => setIsUploading(true));
+  uppy?.on('complete', () => setIsUploading(false));
+
+  uppy?.on('upload-error', handleUploadError(fileMetadataMap));
+
+  uppy?.on('file-removed', handleFileRemoved(fileMetadataMap));
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth="md" fullWidth>
+      <DialogTitle>Upload Attachments</DialogTitle>
+      <DialogContent sx={{ display: 'flex', justifyContent: 'center' }}>
+        {uppy && (
+          <Dashboard
+            uppy={uppy}
+            note={`Files cannot be larger than ${MAX_FILE_SIZE_MB}MB`}
+            proudlyDisplayPoweredByUppy={false}
+            doneButtonHandler={handleClose}
+            theme={theme.palette.mode}
+            metaFields={[
+              {
+                id: 'title',
+                name: 'Title',
+                placeholder: 'Enter file title',
+              },
+              {
+                id: 'name',
+                name: 'Name',
+                placeholder: 'Enter file name',
+              },
```

Comment:
```suggestion
              {
                id: 'file_name',
                name: 'File name',
                placeholder: 'Enter file name',
              },
```
I feel like its a bit clearer to say its the file name, also could this be the first option since its the only required one?